apps/mosaico/src/Article/Article.purs

```diff
@@ -392,3 +396,33 @@ render { props, state, setState } =
         -- Does the string start with wanted element
         isElem elemName elemString =
           Just 0 == String.indexOf (Pattern elemName) elemString
+        
+
+    insertAdsIntoBodyText :: String -> String -> Array BodyElement -> Array BodyElement  
+    insertAdsIntoBodyText contentUnit1 contentUnit2 body =
+      let ad1 = Ad contentUnit1
+          ad2 = Ad contentUnit2
+      in if elements > 15
+        then fromMaybe body $ do
+          bodyWithAd <- insertAt (findAdSpace body $ elements/3) ad1 body
+          insertAt (findAdSpace bodyWithAd $ 2 * elements/3) ad2 bodyWithAd
+        else if elements > 6
+          then fromMaybe body $ insertAt (findAdSpace body $ elements/2) ad1 body
+          else body `snoc` ad1
+      where
+        elements = length body
+        findAdSpace :: Array BodyElement -> Int -> Int
+        findAdSpace body' i 
+          | i > elements = elements
+          | otherwise    = case body' !! (i-1) of
```

Comment:
Would be perhaps a bit neater if we wrote everything as guards:
```
  findAdSpace :: Array BodyElement -> Int -> Int
  findAdSpace body' i
    | i > elements = elements
    | Just (Html _) <- body' !! (i-1)
    , Just (Html _) <- body' !! (i) 
    = i
    | otherwise = findAdSpace body' (i+1)
```

You could of course write that new guard in one line, but usually it's more readable when separating the pattern matching to new lines:
```
| Just (Html _) <- body' !! (i-1), Just (Html _) <- body' !! (i) = i
```

Reply:
Definitely looks cleaner. I just made the change, tested and pushed. So does the comma work like a logical AND there then?

Reply:
Yes, kind of! You can read a bit on guards and pattern guards [here](https://github.com/purescript/documentation/blob/master/language/Pattern-Matching.md#guards)